### PR TITLE
Adds a cooldown to aiming after shooting, also makes shooting while aiming drop the aim

### PR DESCRIPTION
**Pull request by Pacmandevil**
The cooldown is currently 3 seconds as suggested by [this thread](https://forums.aurorastation.org/viewtopic.php?f=18&t=8692&p=81678#p81678)

also adds a nifty proc to Aiming to set said cool-down, it's variable.
The proc is "aim_cooldown(x)" with x being the time you want them unable to aim.

### CODE REVIEW

**Comment by OneOneThreeEight** [comment]
Just out of curiosity, is such a small change something that needs to be tested on dev before it's pushed to master?

**Comment by Lohikar** [comment]
Generally only bugfixes are supposed to target master, and this is not a bugfix.

File: code/modules/projectiles/targeting/targeting_overlay.dm [hunk 1]
```diff
@@ -174,6 +176,10 @@ obj/aiming_overlay/proc/update_aiming_deferred()
 	moved_event.register(owner, src, /obj/aiming_overlay/proc/update_aiming)
 	moved_event.register(aiming_at, src, /obj/aiming_overlay/proc/target_moved)
 	destroyed_event.register(aiming_at, src, /obj/aiming_overlay/proc/cancel_aiming)
+var/aimcooldown //this is hacky im sorry.
```
Comment (skull132) [inline code/modules/projectiles/targeting/targeting_overlay.dm]: Make this `var/static/aimcooldown` in the proc.

**Comment by skull132** [comment]
Technicalities aside. Not sure how good of an idea it is. Because it'll leave the person relying on the autoaim with their pants down if they use it, thus making the mechanic dangerous to use.

**Comment by OneOneThreeEight** [comment]
Is there a way to make the aim intent default to standard shooting after the reflex shot is made, so that the person relying on the autoaim doesn't have the burden of having to hit a button far off target in order to be able to free-fire again?

**Comment by Pacmandevil** [comment]
@OneOneThreeEight There's a hotkey for it I'm pretty sure, J? I'll take a look at it though.

File: code/modules/projectiles/targeting/targeting_overlay.dm (hunk 1 above)
Comment (Pacmandevil) [inline code/modules/projectiles/targeting/targeting_overlay.dm]: Not sure that'll work, as it's used in 
`obj/aiming_overlay/proc/aim_at(var/mob/target, var/obj/thing)
	if (aimcooldown > world.time) ` As well.

File: code/modules/projectiles/targeting/targeting_overlay.dm [hunk 2]
```diff
@@ -174,6 +176,9 @@ obj/aiming_overlay/proc/update_aiming_deferred()
 	moved_event.register(owner, src, /obj/aiming_overlay/proc/update_aiming)
 	moved_event.register(aiming_at, src, /obj/aiming_overlay/proc/target_moved)
 	destroyed_event.register(aiming_at, src, /obj/aiming_overlay/proc/cancel_aiming)
+var/aimcooldown
```
Comment (Lohikar) [inline code/modules/projectiles/targeting/targeting_overlay.dm]: This should not be a global.

**Comment by Pacmandevil** [comment]
@skull132 The Intent already gets changed on shot, I can't get the GUI to update properly though.

File: code/modules/projectiles/targeting/targeting_triggers.dm [hunk 3]
```diff
@@ -20,6 +20,9 @@
 	var/obj/item/weapon/gun/G = aiming_with
 	if(istype(G))
 		G.Fire(aiming_at, owner)
+	cancel_aiming()
+	aim_cooldown(3)
+	owner.aiming.toggle_active()
```
Comment (skull132) [inline code/modules/projectiles/targeting/targeting_triggers.dm]: Just `toggle_active()` should suffice.

**Comment by Pacmandevil** [comment]
Alright, it SHOULD work now. fucking oldcode.

**Pull request merged**